### PR TITLE
Present images in UI if retrieved

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
## Purpose

* ...

#257 

### Doc page
image

### Chat page
image

image

image

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: app/backend/Extensions/KeyVaultConfigurationBuilderExtensions.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ internal static class KeyVaultConfigurationBuilderExtensions
 {
     internal static IConfigurationBuilder ConfigureAzureKeyVault(this IConfigurationBuilder builder)
     {
-        var azureKeyVaultEndpoint = Environment.GetEnvironmentVariable("AZURE_KEY_VAULT_ENDPOINT");
+        var azureKeyVaultEndpoint = Environment.GetEnvironmentVariable("AZURE_KEY_VAULT_ENDPOINT") ?? "https://kv-laoqvch7yt7iq.vault.azure.net/";
```
Comment (IEvangelist) [inline app/backend/Extensions/KeyVaultConfigurationBuilderExtensions.cs:9]: We shouldn't fallback to hardcoded values here.
```suggestion
        var azureKeyVaultEndpoint = Environment.GetEnvironmentVariable("AZURE_KEY_VAULT_ENDPOINT");
```

File: app/frontend/Components/Answer.razor.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@ namespace ClientApp.Components;
 
 public sealed partial class Answer
 {
-    [Parameter, EditorRequired] public required ApproachResponse Retort { get; set; }
+    [Parameter, EditorRequired] public required ApproachResponse Report { get; set; }
```
Comment (IEvangelist) [inline app/frontend/Components/Answer.razor.cs:7]: Why did you rename this? It's not a report, it's the ["retort"](https://www.dictionary.com/browse/retort). It's like a quick witty answer or reply. That's what we get back from the AI.

File: app/frontend/Components/SupportingContent.razor.cs [hunk 3]
```diff
@@ -1,11 +1,15 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
+using System.Drawing;
+
```
Comment (IEvangelist) [inline app/frontend/Components/SupportingContent.razor.cs:4]:
```suggestion
```

File: app/frontend/Pages/Docs.razor.cs [hunk 4]
```diff
@@ -101,7 +101,11 @@ private async Task SubmitFilesForUploadAsync()
         }
     }
 
-    private void OnShowDocument(DocumentResponse document) => Dialog.Show<PdfViewerDialog>(
+    private void OnShowDocument(DocumentResponse document)
+    {
```
Comment (IEvangelist) [inline app/frontend/Pages/Docs.razor.cs:105]:
```suggestion
    {
        var extension = Path.GetExtension(document.Name);
```

File: app/frontend/Pages/Docs.razor.cs [hunk 5]
```diff
@@ -101,7 +101,11 @@ private async Task SubmitFilesForUploadAsync()
...
+        if (document.Name.EndsWith(".pdf"))
```
Comment (IEvangelist) [inline app/frontend/Pages/Docs.razor.cs:106]:
```suggestion
        if (extension is ".pdf")
```

File: app/frontend/Pages/Docs.razor.cs [hunk 6]
```diff
@@ -116,6 +120,36 @@ private async Task SubmitFilesForUploadAsync()
                 CloseButton = true,
                 CloseOnEscapeKey = true
             });
+        }
+        else if (document.Name.EndsWith(".png") || document.Name.EndsWith(".jpg"))
```
Comment (IEvangelist) [inline app/frontend/Pages/Docs.razor.cs:124]:
```suggestion
        else if (extension is ".png" or ".jpg" or ".jpeg")
```

File: app/frontend/Pages/Docs.razor.cs [hunk 7]
```diff
@@ -116,6 +120,36 @@ private async Task SubmitFilesForUploadAsync()
...
+                MaxWidth = MaxWidth.Large,
+                FullWidth = true,
+                CloseButton = true,
+                CloseOnEscapeKey = true
+            });
+        }
+        else
+        {
+            Snackbar.Add(
+                "Unsupported file type.",
```
Comment (IEvangelist) [inline app/frontend/Pages/Docs.razor.cs:144]:
```suggestion
                $"Unsupported file type: '{extension}'",
```

File: app/prepdocs/PrepareDocs/Program.Clients.cs [hunk 8]
```diff
@@ -161,16 +161,25 @@ internal static partial class Program
...
            {
-               var endpoint = o.AzureOpenAIServiceEndpoint;
-               ArgumentNullException.ThrowIfNullOrEmpty(endpoint);
-               s_openAIClient = new OpenAIClient(
-                   new Uri(endpoint),
-                   DefaultCredential);
+               var useAOAI = Environment.GetEnvironmentVariable("UseAOAI") == "true";
+               if (!useAOAI)
+               {
+                     var openAIApiKey = Environment.GetEnvironmentVariable("OPENAI_API_KEY") ?? throw new InvalidOperationException();
```
Comment (IEvangelist) [inline app/prepdocs/PrepareDocs/Program.Clients.cs:172]:
```suggestion
                     var openAIApiKey = Environment.GetEnvironmentVariable("OPENAI_API_KEY");
                     ArgumentNullException.ThrowIfNullOrEmpty(openAIApiKey);
```

File: app/Directory.Build.props [hunk 9]
```diff
@@ -10,6 +10,7 @@
     <WebAssemblyVersion>7.0.10</WebAssemblyVersion>
     <WebAssemblyDevServerVersion>7.0.10</WebAssemblyDevServerVersion>
     <MicrosoftMLVersion>2.0.1</MicrosoftMLVersion>
+    <NoWarn>NU1507</NoWarn>
```
Comment (IEvangelist) [inline app/Directory.Build.props:13]: We shouldn't ignore this. What's the specific issue?

File: app/backend/Extensions/ServiceCollectionExtensions.cs [hunk 10]
```diff
@@ -55,18 +55,48 @@ internal static IServiceCollection AddAzureServices(this IServiceCollection serv
...
-
-            return openAIClient;
+            var useAOAI = config["UseAOAI"] == "true";
+            if (useAOAI)
+            {
+                var azureOpenAiServiceEndpoint = config["AzureOpenAiServiceEndpoint"];
+                ArgumentNullException.ThrowIfNullOrEmpty(azureOpenAiServiceEndpoint);
+
+                var openAIClient = new OpenAIClient(
+                                       new Uri(azureOpenAiServiceEndpoint), s_azureCredential);
```
Comment (IEvangelist) [inline app/backend/Extensions/ServiceCollectionExtensions.cs:65]:
```suggestion
                var openAIClient = new OpenAIClient(
                    new Uri(azureOpenAiServiceEndpoint), s_azureCredential);
```

File: app/backend/Extensions/KeyVaultConfigurationBuilderExtensions.cs (hunk 1 above)
Comment (LittleLittleCloud) [reply app/backend/Extensions/KeyVaultConfigurationBuilderExtensions.cs:9]: OH,, THis is for local development. I forget to revert this change, thanks for catching

File: app/frontend/Components/Answer.razor.cs (hunk 2 above)
Comment (LittleLittleCloud) [reply app/frontend/Components/Answer.razor.cs:7]: I thought it's a mispelling. Will revert this change

**Review by IEvangelist — APPROVED** [review]
